pandas/io/parsers.py

```diff
@@ -1424,7 +1423,13 @@ def __init__(self, src, **kwds):
                               if (i in self.usecols or n in self.usecols)]
 
             if len(self.names) < len(self.usecols):
-                raise ValueError("Usecols do not match names.")
+                bad_cols = [n for n in self.usecols if n not in self.names]
+                if len(bad_cols) > 0:
+                    raise ValueError(("%s specified in usecols but not found "
```

Comment:
@jreback I'm trying to get the following test to work:

```python
msg = ("c, d specified in usecols but not found in names.")
with tm.assertRaisesRegexp(ValueError, msg):
    self.read_csv(StringIO(data), names=['a', 'b'], usecols=['c', 'd'], header=None)
```

Currently this test fails because two exceptions are being raised: one by `parsers.py`, which produces the expected message; and one by `parser.pyx`, which produces a different message. But, when I try to play with this in my interpreter, I only get the first, expected, `ValueError` message. Where does the `parser.pyx`exception get raised to? Is it necessary?